File: 0225-Implement-Stack-using-Queues/0225.py
class MyStack:

    def __init__(self):
        self.Q1, self.Q2 = [], []

    def push(self, x: int) -> None:
        self.Q1.append(x)

    def pop(self) -> int:
        for index, value in enumerate(self.Q1):
            if index == len(self.Q1) - 1:
                last = value
            else:
                self.Q2.append(value)
        self.Q1 = self.Q2
        self.Q2 = []
        return last

    def top(self) -> int:
        return self.Q1[-1]

    def empty(self) -> bool:
        return len(self.Q1) == 0


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()

File: 0225-Implement-Stack-using-Queues/0225.java
import java.util.LinkedList;
import java.util.Queue;

class MyStack {

    private Queue<Integer> Q1;
    private Queue<Integer> Q2;

    public MyStack() {
        this.Q1 = new LinkedList<>();
        this.Q2 = new LinkedList<>();
    }

    public void push(int x) {
        Q1.add(x);
    }

    public int pop() {
        while (Q1.size() > 1) {
            Q2.add(Q1.remove());
        }
        int last = Q1.remove();
        Queue<Integer> temp = Q1;
        Q1 = Q2;
        Q2 = temp;
        return last;
    }

    public int top() {
        while (Q1.size() > 1) {
            Q2.add(Q1.remove());
        }
        int last = Q1.peek();
        Q2.add(Q1.remove());
        Queue<Integer> temp = Q1;
        Q1 = Q2;
        Q2 = temp;
        return last;
    }

    public boolean empty() {
        return Q1.isEmpty();
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */